packages/evolution-common/src/utils/__tests__/ParamsValidatorUtils.test.ts

```diff
@@ -0,0 +1,606 @@
+/*
+ * Copyright 2024, Polytechnique Montreal and contributors
+ *
+ * This file is licensed under the MIT License.
+ * License text available at https://opensource.org/licenses/MIT
+ */
+
+import { ParamsValidatorUtils } from '../ParamsValidatorUtils';
+import { v4 as uuidV4 } from 'uuid';
+
+class TestClass { }
+
+describe('ParamsValidatorUtils', () => {
+
+    describe('isObject', () => {
+        test('should return no errors for a valid object', () => {
+            const errors = ParamsValidatorUtils.isObject('attr', {}, 'TestClass');
+            expect(errors).toEqual([]);
+        });
+
+        test('should return no errors for an undefined value', () => {
+            const errors = ParamsValidatorUtils.isObject('attr', undefined, 'TestClass');
+            expect(errors).toEqual([]);
+        });
+
+        test('should return an error for a non-object value', () => {
+            const errors = ParamsValidatorUtils.isObject('attr', 'invalid', 'TestClass');
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toContain('should be an object');
+        });
+    });
+
+    describe('isRequired', () => {
+        test('should return no errors for a defined value', () => {
+            const errors = ParamsValidatorUtils.isRequired('attr', 'value', 'TestClass');
+            expect(errors).toEqual([]);
+        });
+
+        test('should return an error for an undefined value', () => {
+            const errors = ParamsValidatorUtils.isRequired('attr', undefined, 'TestClass');
+            expect(errors).toHaveLength(1);
+            expect(errors[0].message).toContain('is required');
+        });
+    });
+
+    describe('isInstanceOf', () => {
```

Comment:
Maybe test also with inheriting classes? like another class that inherits TestClass and test all combinations of TestClass/InheritedClass for expected/actual

Reply:
added